app/models/user.rb

```diff
@@ -37,6 +37,15 @@ class User < ApplicationRecord
   validates_acceptance_of :terms_and_conditions
   validates_confirmation_of :password, on: :create
   validates :currency, inclusion: { in: ExchangeRateService::CURRENCIES }
+  validate :email_uniqueness
+
+  protected
+
+  def email_uniqueness
+    if User.all.map(&:email).include?(email) && !persisted?
+      errors.add(:email, :taken)
+    end
+  end
```

Comment:
No problem. I know this is a strange possibility but what if a user signed up using a uniq username but used someone else's email and then proceeded to do things that were unethical. Responses to this new user would go to the other person's email. Somehow, I would like to not have another user on the system have the ability to use one of my emails when they sign up.

I understand your point on not wanting to show someone the email is in use so can you think of another way of handling this?